src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,456 @@
+using LinearAlgebra: diagm, diag, eigen, eigvals, eigvecs, Symmetric, Diagonal, factorize, tr, norm, cholesky, LowerTriangular
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) =
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the manifold $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n), N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = div(N*(N+1), 2)
+
+@doc doc"""
+    LinearAffineMetric <: Metric
+
+The linear affine metric is the metric for symmetric positive definite matrices, that employs
+matrix logarithms and exponentials, which yields a linear and affine metric.
+"""
+struct LinearAffineMetric <: RiemannianMetric end
+# Make this metric default, i.e. automatically convert
+convert(::Type{SymmetricPositiveDefinite{N}},M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}) where N = M.manifold
```

Comment:
If you do this convert:
`convert(::Type{SymmetricPositiveDefinite{N}},M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}) where N = M.manifold`
adding 
`convert(::Type{SymmetricPositiveDefinite},M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}) where N = M.manifold`
would also make sense (people won't have to specify dimension when it can be inferred from the converted manifold).

The second `convert` below is OK, the `N` is buried much deeper so I'd leave it as is.

Reply:
I had hoped my two were enough. Can you explain why this additional one is better? Or can I then leave the first one out, since N can be infered anyways? I am not yet that much of an expect of `convert`s...

Reply:
It's not very important, just a small convenience. With the code you currently have this:
`convert(SymmetricPositiveDefinite{3}, M)`
would work (as long `M` is a `MetricManifold{SymmetricPositiveDefinite{3},LinearAffineMetric}`)
but 
`convert(SymmetricPositiveDefinite, M)`
would not.
With the additional convert I've written both would work.

You can't leave the `N` in the first argument out completely because `convert(::Type{SymmetricPositiveDefinite}, ...)` wouldn't allow `convert(SymmetricPositiveDefinite{3}, M)` (with dimension explicitly written) and `convert(::Type{<:SymmetricPositiveDefinite}, ...)` would allow non-matching dimensions. `StaticArrays` has a pretty well designed conversions, a similar pattern is, for example, here: https://github.com/JuliaArrays/StaticArrays.jl/blob/a3bca3593c8ea824ebc7c95c37c6b38f52568254/src/SizedArray.jl#L70

Reply:
Thanks for the explanation.